commons-vfs2/src/main/java/org/apache/commons/vfs2/impl/DefaultFileSystemManager.java

```diff
@@ -712,13 +713,49 @@ public FileObject resolveFile(final FileObject baseFile, final String uri, FileS
             // An absolute URI - locate the provider
             final FileProvider provider = providers.get(scheme);
             //In the case of SFTP set the path from root if the param is presented in URL
-            if (provider instanceof SftpFileProvider && "true".equals(queryParam.get(SftpConstants
-                    .SFTP_PATH_FROM_ROOT))) {
+            if (provider instanceof SftpFileProvider) {
                 if (fileSystemOptions == null) {
                     fileSystemOptions = new FileSystemOptions();
                 }
-                ((SftpFileSystemConfigBuilder) ((provider).getConfigBuilder())).setUserDirIsRoot
-                        (fileSystemOptions, false);
+                if ("true".equals(queryParam.get(SftpConstants.SFTP_PATH_FROM_ROOT))) {
+                    ((SftpFileSystemConfigBuilder) (((SftpFileProvider) provider).getConfigBuilder()))
+                            .setUserDirIsRoot(fileSystemOptions, false);
+                }
+
+                String proxyHost = queryParam.get(SftpConstants.PROXY_SERVER);
+
+                if (proxyHost != null && !proxyHost.isEmpty()) {
+                    String proxyPortStr = queryParam.get(SftpConstants.PROXY_PORT);
+                    int proxyPort;
+                    try {
+                        proxyPort = Integer.parseInt(proxyPortStr);
+                    } catch (NumberFormatException | NullPointerException e) {
+                        proxyPort = 8080;
+                    }
+                    String proxyUser = queryParam.get(SftpConstants.PROXY_USERNAME);
+                    String proxyPassword = queryParam.get(SftpConstants.PROXY_PASSWORD);
+                    String proxyType = queryParam.get(SftpConstants.PROXY_TYPE);
+
+                    if (SftpConstants.SOCKS.equals(proxyType)) {
+                        ((SftpFileSystemConfigBuilder) (provider.getConfigBuilder())).setProxyType(fileSystemOptions,
+                                SftpFileSystemConfigBuilder.PROXY_SOCKS5);
+                    } else {
+                        if (proxyType != null && !proxyType.isEmpty() && !SftpConstants.HTTP.equals(proxyType)) {
```

Comment:
Shall we use isNotBlank(proxyType) here? It will check both != null and .isEmpty(). So no need of having 2 conditions. Just a suggestion. :) In R753 also.